### PR TITLE
add: redirect for inline auth 401 error 🪃❌

### PR DESCRIPTION
**Pull request by wflore19**
Closes [#244](https://github.com/trypear/pearai-app/issues/244)

## Description

- Add new error message window w/ redirect to PearAI login action.

inline-auth-redirect

## Checklist

- [x] The base branch of this PR is `preview`, rather than `main`

### CODE REVIEW

**Comment by wflore19** [comment]
## Update

Removed the dupliate word 'Error'

new error message
image

**Review by cqdev-co — COMMENTED** [review]
Good work! Is there anyway to hide the link to the server and make it something else? like "PearAI Server"

**Comment by wflore19** [comment]
> Good work! Is there anyway to hide the link to the server and make it something else? like "PearAI Server"

I can hard code it into the error message but this is what the API endpoint send in response to an Auth error.

File: extensions/vscode/src/diff/verticalPerLine/manager.ts [hunk 1]
```diff
@@ -223,13 +219,25 @@ export class VerticalPerLineDiffManager {
           onlyOneInsertion,
         ),
       );
-    } catch (e) {
-      console.error("Error streaming diff:", e);
-      vscode.window.showErrorMessage(`Error streaming diff: ${e}`);
+    } catch (error) {
+      vscode.window
+        .showErrorMessage(
+          `PearAI access requires login: Fetch Error: Invalid response body while trying to fetch PearAI Server`,
```
Comment (nang-dev) [inline extensions/vscode/src/diff/verticalPerLine/manager.ts:225]: Yo we should only do this if we are sure it is a login issue.

there are other issues that can arise

so it should like "if error.message includes XXXXXX then do this"
else 
console.error("Error streaming diff:", e);
      vscode.window.showErrorMessage(`Error streaming diff: ${e}`);

Comment (nang-dev) [reply extensions/vscode/src/diff/verticalPerLine/manager.ts:225]: @wflore19

**Comment by wflore19** [comment]
Fixed merge conflicts @nang-dev @Fryingpannn

**Review by Fryingpannn — APPROVED** [review]
Noicee

**Pull request merged**